### PR TITLE
Release 24.08rc1

### PR DESCRIPTION
**Pull request by ShahanaFarooqui**
Release candidate 1 for v24.08.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,135 @@ All notable changes to this project will be documented in this file.
...
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+
+## [24.08rc1] - 2024-08-12: "TO BE ADDED"
+
+This release named by @Lagrang3.
+
+### Added
+
+ - protocol: pay can now pay to bolt12 invoices if entry to blinded hop is specified as a short_channel_id (rather than node id). ([#7461])
```
Comment (rustyrussell) [inline CHANGELOG.md:13]: This is EXPERIMENTAL...

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,135 @@ All notable changes to this project will be documented in this file.
...
+ - protocol: onion messages are now supported by default. ([#7455])
+ - protocol: onion messages can now be forwarded by short_channel_id. ([#7455])
+ - offers: automatically add a blinded path from a peer if we have no public channels, so unannounced nodes can have offers too. ([#7461])
```
Comment (rustyrussell) [inline CHANGELOG.md:16]: This too

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,135 @@ All notable changes to this project will be documented in this file.
...
+ - offers: we can now self-fetch and self-pay BOLT12 offers and invoices. ([#7461])
```
Comment (rustyrussell) [inline CHANGELOG.md:17]: this too

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,135 @@ All notable changes to this project will be documented in this file.
...
+ - pyln-client: implement setconfig hook for plugins so you can see changes in `dynamic` options. ([#7289])
+ - doc: all examples in the documentation are now generated from running the code, so they are current. ([#7457])
+
+
+### Changed
+
+ - protocol: We now send current peers our changed gossip (even if they set timestamp_filter otherwise), not just on reconnect. ([#7554])
+ - protocol: we now always ask the first peer for all its gossip. ([#7512])
+ - connectd: I/O optimizations to significantly speed up larger nodes. ([#7365])
+ - connectd: now should use far less CPU on large nodes. ([#7365])
```
Comment (rustyrussell) [inline CHANGELOG.md:40]: Maybe move this under Fixed?

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,135 @@ All notable changes to this project will be documented in this file.
...
+
+ - JSON-RPC: `sendonionmessage` (was experimental only, use `injectonionmessage`) ([#7461])
+ - JSON-RPC: `autocleaninvoice` command (deprecated v22.11, EOL v24.02) ([#7298])
+ - plugins: `estimatefees` returning feerates by name (e.g. "opening"); deprecated in v23.05. ([#7380])
+
+
+### Fixed
+
+ - protocol: we can now open unannounced channels with LND nodes again. ([#7564])
+ - plugins: pay can now pay a bolt12 invoice even if we, ourselves, are the head of the blinded path within it. ([#7461])
```
Comment (rustyrussell) [inline CHANGELOG.md:76]: EXPERIMENTAL

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,135 @@ All notable changes to this project will be documented in this file.
...
+ - renepay: finalized routes have to be processed and ([#7357])
```
Comment (rustyrussell) [inline CHANGELOG.md:77]: and.. .?

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,135 @@ All notable changes to this project will be documented in this file.
...
+ - config: whitespace at the end of (most) options is now ignored, not complained about. ([#7251])
```
Comment (rustyrussell) [inline CHANGELOG.md:78]: Move this higher?

File: CHANGELOG.md (hunk 6 above)
Comment (rustyrussell) [reply CHANGELOG.md:77]: also, renepay is EXPERIMENTAL

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,135 @@ All notable changes to this project will be documented in this file.
...
+
+ - protocol: pay can now pay to bolt12 invoices if entry to blinded hop is specified as a short_channel_id (rather than node id). ([#7461])
+ - protocol: onion messages are now supported by default. ([#7455])
+ - protocol: onion messages can now be forwarded by short_channel_id. ([#7455])
+ - offers: automatically add a blinded path from a peer if we have no public channels, so unannounced nodes can have offers too. ([#7461])
+ - offers: we can now self-fetch and self-pay BOLT12 offers and invoices. ([#7461])
+ - pay: The pay plugin now checks whether we have enough spendable capacity before computing a route, returning a clear error message if we don't ([#7418])
+ - pay: Payments now emit `channel_hint_updated` notification to share inferred balances and observations across multiple payments. ([#7487])
+ - pay: The pay plugin now returns better error codes ([#7418])
+ - renepay: prune the network by disabling channels we don't like, eg. very low max_htlc. ([#7403])
```
Comment (rustyrussell) [inline CHANGELOG.md:21]: EXPERIMENTAL

**Review by rustyrussell — COMMENTED** [review]
Mainly just sections moves, for things which are Experimental (offers, renepay)

File: CHANGELOG.md (hunk 6 above)
Comment (Lagrang3) [reply CHANGELOG.md:77]: Also the message was truncated. May be change to this
```suggestion
 - renepay: bugfix, un-reserve routes that have been completed or failed ([#7357])
```

Comment (ShahanaFarooqui) [reply CHANGELOG.md]: Changed the message as suggested.

**Comment by ShahanaFarooqui** [comment]
ACK b2414378fc111b3771c4e461778856bd0ec1f221

**Pull request merged**